core/src/main/java/com/netflix/conductor/core/execution/DeciderService.java

```diff
@@ -315,33 +304,36 @@ void updateWorkflowOutput(final Workflow workflow, Task task) {
             return;
         }
 
-        Optional<Task> terminateTask = allTasks.stream()
-            .filter(t -> TaskType.TERMINATE.name().equals(t.getTaskType()) && t.getStatus().isTerminal()
-                && t.getStatus().isSuccessful())
-            .findFirst();
-        if (terminateTask.isPresent()) {
-            if (!terminateTask.get().getOutputData().isEmpty()) {
-                workflow.setOutput(terminateTask.get().getOutputData());
+        Optional<Task> optionalTask = allTasks.stream()
+                .filter(t -> TaskType.TERMINATE.name().equals(t.getTaskType()) && t.getStatus().isTerminal()
+                        && t.getStatus().isSuccessful())
+                .findFirst();
+        if (optionalTask.isPresent()) {
+            Task terminateTask = optionalTask.get();
+            if (StringUtils.isNotBlank(terminateTask.getExternalOutputPayloadStoragePath())) {
```

Comment:
Should this be applied to all sync system tasks?

Reply:
other sync system tasks are handled in the else block below, here we need special handling of the terminate task because terminate task can be used to override workflow output.